### PR TITLE
Use generator instead of list expand or add method

### PR DESCRIPTION
**Pull request by jmishra01**
We can use generator to create pretty string of Tree in place of list. This PR reduce memory usage.

### CODE REVIEW

**Comment by erezsh** [comment]
I like this change.

You can turn a few more of the "yield"s to "yield from"s.

**Comment by jmishra01** [comment]
@erezsh Done!

**Comment by erezsh** [comment]
What I meant is that you don't need any for loop..

```python
yield from (i for i in ('\t', '%s' % (self.children[0],), '\n'))
```

can be

```python
yield from ('\t', '%s' % (self.children[0],), '\n')
```

etc.

**Comment by jmishra01** [comment]
Thank you for review, I change few lines,  use yield with f-string method.

File: lark/tree.py [hunk 1]
```diff
@@ -149,13 +148,15 @@ def iter_subtrees_topdown(self):
         Iterates over all the subtrees, return nodes in order like pretty() does.
         """
         stack = [self]
+        stack_append = stack.append
```
Comment (erezsh) [inline lark/tree.py:151]: I assume you added this for performance. Did you benchmark it on a realistic use-case?

Comment (erezsh) [reply lark/tree.py:151]: Did you try also changing `Tree` to a local variable (`_Tree` or such)?

Comment (jmishra01) [reply lark/tree.py:151]: Not such type of thing, but I create child class of it, and enhance some part for better performance. 
Also working on multi-threading, with cythonise some part of code and logic.

Comment (erezsh) [reply lark/tree.py:151]: I meant specifically inside this function, to skip the global lookup.

Comment (jmishra01) [reply lark/tree.py:151]: > Also, you're aware of https://github.com/lark-parser/lark_cython?

Yes. 
I recommend, we should start implement lark in C++ or Rust, so with WASM and  Cython (or Pybind11), we can use optimise version of it in Python and JS world.

Comment (jmishra01) [reply lark/tree.py:151]: I'm interest to take both the tasks, if you allow.

**Comment by erezsh** [comment]
Thanks!

File: lark/tree.py (hunk 1 above)
Comment (jmishra01) [reply lark/tree.py:151]: Sure, I will take care of it.

**Pull request merged**